internal/pkg/filesystem/filesystem.go

```diff
@@ -0,0 +1,330 @@
+package filesystem
```

Comment:
**Všetky metody potrebne pre pracu s `filesystem` som dal do tohto balicka.**
Kedze sa blbo v kode pracovalo a uz ma to hnevalo.

Ma to tieto vyhody:
- Logovanie FS operacii na jednom mieste.
- `localfs` (vid dalej) generuje vsetky cesty od `projectDir` - base path, ... a teda nemusi vsade rucne generovat absolutnu cestu -> mensie riziko chyby a toho ze prepisem nieco mimo projektu.
- `memoryfs` - backed pre filesystem implementovany iba v pamati, ... vhodne pre unit testy + mozno sa zide ak by sme to chceli skompilovat do WASM a pouzit v piamo v UI/prehlidaci.


**V tomto PR pridavam iba tuto kniznicu. Nemenim pracu so subormi v ostatnom kode -> to bude dalsi PR.
Viac sa to osekat nedalo, su tu rozsiahle testy.**